org.lflang/src/org/lflang/generator/ReactionInstance.java

```diff
@@ -448,6 +448,57 @@ public String toString() {
         return getName() + " of " + parent.getFullName();
     }
 
+    /**
+     * Determine logical execution time for each reaction during compile
+     * time based on immediate downstream logical delays (after delays and actions)
+     * and label each reaction with the minimum of all such delays.
+     */
+    public TimeValue getLogicalExecutionTime() {
+        if (this.let != null) {
+            return this.let;
+        }
+
+        if (this.parent.isGeneratedDelay()) {
+            return this.let = TimeValue.ZERO;
+        }
+
+        TimeValue let = null;
+
+        // Iterate over effect and find minimum delay.
+        for (TriggerInstance<? extends Variable> effect : effects) {
+            if (effect instanceof PortInstance) {
+                var afters = this.parent.getParent().children.stream().filter(c -> {
+                    if (c.isGeneratedDelay()) {
+                        return c.inputs.iterator().next().getDependsOnPorts().iterator().next().instance
+                                       .equals((PortInstance) effect);
+                    }
+                    return false;
+                }).map(c -> c.actions.iterator().next().getMinDelay())
```

Comment:
I might just be unfamiliar with the `.iterator().next()` pattern. Could you clarify what it does?

Reply:
In the 'Delay' reactor, there is only one delay reaction, so we simply used the first iteration.
Is it the right explanation? @CloverCho

Reply:
Ah, okay. I think `.get(0)` is what we've used most often.

Reply:
Thank you for your nice programming advice!
I modified it in https://github.com/lf-lang/lingua-franca/pull/1292/commits/ff0782633922481312131e2321730ef447577194.